### DISCUSSION TITLE
How to detect app is minimize or close state?

### DISCUSSION THREAD

**Discussion by rahulmishra1991** [microsoft/react-native-windows#12288] [Q&A] [answered]
I need to manage app on based on minimize or close app states. I have added AppState on change eventListner. And based on the AppState change minimize or close I need to trigger 2 different actions.
But Minimize and Close both states are same background so I am not able to differentiate and trigger actions according to my requirement.

Here is my code:

AppState.addEventListener('change', this._handleAppStateChange);

 handleAppStateChange = async (appState) => {
    console.log(appState)
  };

In both the conditions app is returning background only.

Anyone help here for windows app only.

**Comment by jonthysell** [comment, marked as answer]
It looks like the RN AppState API doesn't really consider that: https://reactnative.dev/docs/appstate#app-states

You might consider writing your own native module to crate your own JS events for this state change and have them fire when the native close/minimize event happens.

**Reply by rahulmishra1991** [reply]
Thanks for the response @jonthysell.

Yes I got that but I am very new in windows app development and not able to create own native module.

**Reply by chrisglein** [reply]
It's possible you can handle this from JS with [react-native-winrt](https://github.com/microsoft/react-native-winrt). It exposes Windows APIs to JS without you having to create your own native module. I can't say with confidence that this API is covered there, but you can do some looking?
Also, for reference, here are the docs for creating your own native module: https://microsoft.github.io/react-native-windows/docs/native-modules
